pkg/storage/formattedstore/formattedstore.go

```diff
@@ -122,14 +168,31 @@ func (s *formatStore) Get(k string) ([]byte, error) {
 		return nil, fmt.Errorf(failParseFormattedValue, err)
 	}
 
+	// put data in cache to retrieve it next time
+	if s.cacheStore != nil {
+		if err := s.cacheStore.Put(k, value); err != nil {
+			return nil, err
+		}
+	}
+
 	return value, nil
+
 }
 
 // The "don't filter" switch is here since it's possible for the underlying storage to have already done
 // the filtering, in which case it would be redundant to do it here too.
 // TODO (#2315) While doing the storage interface rework, see if the skipIteratorFiltering switch is still
 //  needed.
 func (s *formatStore) Iterator(startKey, endKey string) storage.StoreIterator {
+	if s.cacheStore != nil {
+		r := s.cacheStore.Iterator(startKey, endKey)
+		if r.Error() == nil {
+			return r
```

Comment:
Can't return `r` immediately - need to run it through the loop below

Reply:
no loop needed

Reply:
Wait - my comment is only true if the cache store is storing encrypted data. If the cache store contains the decrypted stuff already, then I think you're ok.